### DISCUSSION TITLE
Webhook calling failed while creating shovel and federation through manifests

### DISCUSSION THREAD

**Discussion by paywagh11** [rabbitmq/rabbitmq-server#9521] [Questions]
I have 2 rabbitmq cluster setup in 2 different namespace in AKS cluster.
I have created Federation and shovel through RMQ UI and everything was working fine.

But when I tried using yaml files, I wasn't able to create both shovel and federation.
After applying secret for both federation and shovel, when I tried applying federation.yaml and shovel.yaml, it shows failed calling webhook error.

Please find below yaml files for shovel which I have applied:
1. Shovel_secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: shovel-secret
type: Opaque
data:
  destUri: # encoded value for destination uri (amqp://defaultusername:defaultpassword@host)
  srcUri: # encoded value for SRC uri (amqp://defaultuser@)

2. shovel.yaml
apiVersion: rabbitmq.com/v1beta1
kind: Shovel
metadata:
  name: shovel-example
  namespace: rmq-int
spec:
  name: "shovel-example"
  uriSecret:
    name: shovel-secret
  srcQueue: "test.source"
  srcConsumerArgs:
    x-priority: 
  destQueue: "test.target"
  destPubkishPropertirs:
    delivery_mode: 2
  rabbitmqClusterReference:
   name: rabbitmqcluster-external

I applied these files in rmq-int namespace. Secret got created but for shovel I got below error:
Internal error: error when creating "shovel.yaml": Internal error occurred: failed calling webhook "vshovel.kb.io": failed to call webhook: Post "https://webhook-service.rabbitmq-internal.svc:443/validate-rabbitmq-com-v1beta1-shovel?timeout=10s": tls: failed to verify certification: x509: certificate signed by unknown authority.

Also modified the secret uri for SRC and target queue in same rmq cluster and got below error:
Internal error: error when creating "shovel.yaml": Internal error occurred: failed calling webhook "vshovel.kb.io": failed to call webhook: Post "https://webhook-service.rabbitmq-internal.svc:443/validate-rabbitmq-com-v1beta1-shovel?timeout=10s": EOF

Can someone please suggest how I can apply shovel in my rmq cluster?
Or if there is something wrong with my yaml files?
Also tried creating it in bitnami rabbitmq cluster as well and got above errors.

**Comment by michaelklishin** [comment]
@paywagh11 it's not clear to me how do you deploy RabbitMQ to Kubernetes, e.g. whether you use our cluster operator, or DIY.

> certificate signed by unknown authority

means that something (a Kubernetes component) fails to [perform peer certificate chain verification](https://rabbitmq.com/ssl.html#peer-verification) on an endpoint it calls (a RabbitMQ HTTP API node?)

We cannot know what that may be, as we have neither the Kubernetes deployment nor any node log files to work with. We do not guess in this community.

**Reply by paywagh11** [reply]
@michaelklishin thanks for the reply. 
I am deploying rmq using official(i.e. given in this GitHub repo)  rabbitmq operator in kubernetes.

Just, we have created certificate for messaging topology operator and have been using the same. It's working fine if we create any topology objects using rmq UI but not through manifest files.

**Comment by Zerpet** [comment]
What versions of Cluster Operator and Topology Operator are you using?

One thing that catches the eye is the address of the webhook `https://webhook-service.rabbitmq-internal.svc` was changed in recent versions to something more specific that `webhook-service`.

Are you able to create other topology objects in your cluster e.g. a queue?

**Reply by paywagh11** [reply]
@Zerpet , thanks for the reply.

I am using versions as below:
Cluster operator- 2.3.0
Msg topology operator - 1.11.0

Actually I am not able to create any exchange or queue as well in my cluster using manifest files.

Could you please let me know where I need to fix things.

**Reply by Zerpet** [reply]
Make sure you follow the instructions described [here](https://rabbitmq.com/kubernetes/operator/install-topology-operator.html), in section `Install with Generated Certificates`. You are probably missing a certificate somewhere in the configuration.
